nilearn/_utils/niimg.py

```diff
@@ -30,6 +31,26 @@ def _safe_get_data(img):
     return img.get_data()
 
 
+def _get_data_dtype(img):
+    try:
+        return img.get_data_dtype()
+    except AttributeError:
+        return img.get_data().dtype
+
+
+def _get_target_dtype(dtype, target_dtype):
```

Comment:
I think that what you are looking for is somethink like dtype.kind, which is 'f' for floating point types, and 'i' for integer types.